src/profiler.cpp

```diff
@@ -364,12 +364,14 @@ int Profiler::convertNativeTrace(int native_frames, const void** callchain, ASGC
                 // as Java frames returned by AGCT. Terminate the scan here.
                 return depth;
             } else if (mark == MARK_COMPILER_ENTRY && _features.comp_task) {
-                // Insert current compile task as a pseudo Java frame
-                jmethodID compile_task = getCurrentCompileTask();
-                if (compile_task != NULL) {
-                    frames[depth].bci = 0;
-                    frames[depth].method_id = compile_task;
-                    depth++;
+                if (depth > 0 && strstr((const char*) frames[depth-1].method_id, "14compile_method") != NULL) {
```

Comment:
Alternatively, can we mark `Compiler::compile_method` and `C2Compiler::compile_method` as compilation entry points instead of `invoke_compiler_on_method`? We won't need this strstr check then. As a side effect, it will be easy to see on a flamegraph total time spent in C1 vs. C2.